script.sh

```diff
@@ -147,7 +146,41 @@ case "${TRAVIS_DEBIAN_DISTRIBUTION}" in
 		;;
 esac
 
-# TRAVIS_DEBIAN_DISTRIBUTION is now set, so set dependent options
+# detect derivatives
+if [ "${TRAVIS_DEBIAN_DERIVATIVE:-}" = "" ]
+then
+	# strip derivate common pockets
+	for suffix in proposed updates security
+	do
+		TRAVIS_DEBIAN_DISTRIBUTION="${TRAVIS_DEBIAN_DISTRIBUTION%%-$suffix}"
```

Comment:
Can you give an example (in the comment) when/why this might be needed? Is it Ubuntu specific?

Reply:
We have branch name matching the state of the various "pockets" (sorry, this is the real term) to mirror what's in the archive.
In ubuntu, everything goes to $distroname-proposed before migrating (via britney) to $distroname which is the release pocket. Same for $distroname-proposed to $distroname-updates once a release is out.
This is a little bit like sid -> unstable -> stable.

The correct term would thus be "distribution", but it might confuse with the other distribution name.
I guess you are right, I'm going to remove this and replace with precise*|trusty* and so on, as this is only used to set TRAVIS_DEBIAN_DERIVATIVE.
Then, based on TRAVIS_DEBIAN_DERIVATIVE, I can strip those common suffix (and use the term "pocket" thus! ;)) to get the correct TRAVIS_DEBIAN_DISTRIBUTION, even if we are on a -proposed branch name.